boardswarm-cli/src/main.rs

```diff
@@ -386,6 +386,54 @@ struct BmapWriteArgs {
     file: PathBuf,
 }
 
+#[derive(Clone, Debug)]
+enum VolumeArg {
+    Id(u64),
+    Name(String),
+}
+
+impl VolumeArg {
+    async fn volume(&self, mut client: Boardswarm) -> Result<Option<u64>, anyhow::Error> {
+        match self {
+            VolumeArg::Id(id) => Ok(Some(*id)),
+            VolumeArg::Name(name) => {
+                let mut items = client.list(ItemType::Volume).await?;
+                items.retain(|i| i.name.starts_with(name));
```

Comment:
I thought the name contained the "@something" part, in which case instances would be taken care of: If the prefix is unique enough then it is a match. E.g. if there are

ttyUSB0
ttyUSB0@another-instance

then "ttyUSB0" is ambiguous if "ttyUSB0" and "ttyUSB0@another-instance" are the names. OTOH if there are e.g.

ttyUSB1
ttyUSB0@another-instance

then "ttyUSB0" _is_ unique enough.

The code, after matching on a prefix, counts how many matches there are. If there's just 1, then it's a perfect match, if there's 0 then nothing matched (so error) and if there's > 1 match then it's an error, too, because the user-provided name is ambiguous.

Reply:
boardswarm doesn't generate `@<instance>` names though ;) Also no names aren't guaranteed to be unique the primary key is the id, the name is just a property